<!-- edge_detection_service_test.launch-->

<launch>
  <!-- Launch argument for the bag file path -->
  <arg name="bag_path" default="/root/catkin_ws/src/edge_detection/data/bagfiles/withpointcloud.bag"
       doc="Absolute path to the .bag file you want to play" />

  <!-- Use simulated time -->
  <param name="use_sim_time" value="true" />

  <!-- Edge detection service node -->
  <node pkg="edge_detection" type="edge_detection_service.py" name="edge_detection_service" output="screen">
    <!-- Choose the method: "canny", "sobel", "laplacian", "prewitt", "roberts", or "hed" -->
    <param name="method" value="canny" />
    <param name="canny_low_threshold"  value="250" />
    <param name="canny_high_threshold" value="300" />
    <param name="camera_info_topic" value="/camera/depth/camera_info" />

  </node>

  <node pkg="edge_detection" type="edge_detection_client_test.py" name="edge_detection_client_test" output="screen">
  </node>
</launch>
